src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6842,21 +6855,53 @@ private BoundExpression ConvertToArrayIndex(BoundExpression index, SyntaxNode no
             return result;
         }
 
+        private BoundExpression TryImplicitConversionToArrayIndex(BoundExpression expr, WellKnownType wellKnownType, SyntaxNode node, DiagnosticBag diagnostics)
+        {
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            TypeSymbol type = GetWellKnownType(wellKnownType, ref useSiteDiagnostics);
```

Comment:
>useSiteDiagnostics [](start = 66, length = 18)

Should the `useSiteDiagnostics` be added to `diagnostics` if `result != null`?

Reply:
I don't think so. If we attempted to find System.Index but couldn't, we should get an error about not being able to convert to int, not an error about System.Index not being found. Basically, the diagnostics are only useful if we actually use that type as the target of the conversion.

Reply:
But if we do find `System.Index`, shouldn't we include any use-site diagnostics (perhaps just warnings in that case)?

---
In reply to: [226712363](https://github.com/dotnet/roslyn/pull/30618#discussion_r226712363) [](ancestors = 226712363)